p4_symbolic/util/status.h

```diff
@@ -0,0 +1,37 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This is a test file for our protobuf specifications of bmv2 json.
+// It reads an input bmv2 json string (usually the output of p4c) via stdin,
+// it parses the string using protobuf, and then dumps the parsed protobuf
+// objects using protobuf text format and json.
+// The dumps are written to output files whose paths are provided as command
+// line arguments.
+
+#ifndef P4_SYMBOLIC_UTIL_STATUS_H_
+#define P4_SYMBOLIC_UTIL_STATUS_H_
+
+#include "absl/status/status.h"
+#include "google/protobuf/stubs/status.h"
+
+namespace p4_symbolic {
+namespace util {
+
+// Transform protobuf status into semantically equivalent absl status.
```

Comment:
there is a github issue, and they responded to it by saying they do not want to do it, and want to keep the enums separate indefinitely  and not have any absl in the API. I will put a link.

I could have written this as a cast, currently all the fields in protobuf exist in absl with the same int value. I was not sure if this will continue to be the case, and decided to do it this way instead (obviously I did not write this code, I generated it from a python script)

Reply:
Actually the issue is not super related to this, so I will not add it as a comment. But in the discussion on this issue, the idea of migrating to absl::Status is discussed
https://github.com/protocolbuffers/protobuf/issues/1404

The relevant piece:
"Ultimately we may not be able to migrate to absl::Status anyway, because we expose Status in our public API. At the moment we have decided not to expose ABSL types directly in our public API. So our decision for now is to keep our own copy/port of Status indefinitely."

Reply:
Thanks for looking into it. I agree that the explicit conversion is better, using a cast seems very brittle.